pkg/ui/src/js/defs.bzl

```diff
@@ -0,0 +1,108 @@
+"implementation is borrowed from https://github.com/bazelbuild/rules_nodejs/blob/stable/examples/protobufjs/defs.bzl"
```

Comment:
I think this docstring pattern isn't really used anywhere else in tree? You can just replace this with a comment.